drivers/regulator/regulator_axp2101.c

```diff
@@ -0,0 +1,463 @@
+/*
+ * Copyright (c) 2024 Lothar Felten <lothar.felten@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT x_powers_axp2101_regulator
+
+#include <errno.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/sys/linear_range.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_instance.h>
+
+LOG_MODULE_REGISTER(regulator_axp2101, CONFIG_REGULATOR_LOG_LEVEL);
+
+/* Output control registers */
+#define AXP2101_REG_DCDC12345_CONTROL 0x80U
+#define AXP2101_REG_DCDC1_VOLTAGE     0x82U
+#define AXP2101_REG_DCDC2_VOLTAGE     0x83U
+#define AXP2101_REG_DCDC3_VOLTAGE     0x84U
+#define AXP2101_REG_DCDC4_VOLTAGE     0x85U
+#define AXP2101_REG_DCDC5_VOLTAGE     0x86U
+#define AXP2101_REG_LDOGRP1_CONTROL   0x90U
+#define AXP2101_REG_LDOGRP2_CONTROL   0x91U
+#define AXP2101_REG_ALDO1_VOLTAGE     0x92U
+#define AXP2101_REG_ALDO2_VOLTAGE     0x93U
+#define AXP2101_REG_ALDO3_VOLTAGE     0x94U
+#define AXP2101_REG_ALDO4_VOLTAGE     0x95U
+#define AXP2101_REG_BLDO1_VOLTAGE     0x96U
+#define AXP2101_REG_BLDO2_VOLTAGE     0x97U
+#define AXP2101_REG_CPUSLDO_VOLTAGE   0x98U
+#define AXP2101_REG_DLDO1_VOLTAGE     0x99U
+#define AXP2101_REG_DLDO2_VOLTAGE     0x9AU
+
+struct regulator_axp2101_desc {
+	const uint8_t enable_reg;
+	const uint8_t enable_mask;
+	const uint8_t enable_val;
+	const uint8_t vsel_reg;
+	const uint8_t vsel_mask;
+	const uint8_t vsel_bitpos;
+	const int32_t max_ua;
+	const uint8_t num_ranges;
+	const struct linear_range *ranges;
+};
+
+struct regulator_axp2101_data {
+	struct regulator_common_data data;
+};
+
+struct regulator_axp2101_config {
+	struct regulator_common_config common;
+	const struct regulator_axp2101_desc *desc;
+	const struct i2c_dt_spec i2c;
+
+	LOG_INSTANCE_PTR_DECLARE(log);
```

Comment:
Is instance level logging really needed? Very few drivers do this, and, in most cases, I'd say there's only a single regulator instance on a system.

Reply:
Hi @gmarull,
I remove the per instance logging.